### DISCUSSION TITLE
Field name with a . inside

### DISCUSSION THREAD

**Discussion by frack113** [SigmaHQ/pySigma#27] [Q&A] [answered]
Hi,
When I was checking the Zeek rules, I changed a field to its corect value "id.orig_h" .
Although in the detection it works without problem, in the "condition" part some backend crashes...

Is there this problem of field names with a "." too ?
Thanks

**Comment by thomaspatzke** [comment, marked as answer]
Dots in field names shouldn't be a problem. Where does the problem occurs exactly?

**Reply by frack113** [reply]
when correct zeek_dce_rpc_domain_user_enumeration.yml with correct field name [SigmaHQ  log](https://github.com/SigmaHQ/sigma/runs/4202713607?check_suite_focus=true)
```yaml
    timeframe: 30s
    condition: selection | count(operation) by id.orig_h > 4
```
get
```bash
Error: Sigma parse error in rules/network/zeek/zeek_dce_rpc_domain_user_enumeration.yml: Unexpected token in condition at position .orig_h > 4
...
...
An unsupported feature is required for this Sigma rule (rules/linux/modsecurity/modsec_mulitple_blocks.yml): Aggregations not implemented for this backend
make: *** [Makefile:26: test-sigmac] Error 4
Error: Process completed with exit code 2.
```

**Reply by thomaspatzke** [reply]
Ah, ok...dots are missing in the token definition in the expression parser. This is not a pySigma issue but one in the legacy sigmatools. I try to fix this in the next week.

**Reply by frack113** [reply]
Sorry , yes it is sigmac. The question is to know if pySigma have the same trouble.

**Reply by thomaspatzke** [reply]
No, the syntax will be dropped in favor of correlations syntax.

**Reply by thomaspatzke** [reply]
I'm currently working on the fix and just discovered that the mapping to `id.orig_h` is done by the Zeek configurations, e.g. `tools/config/splunk-zeek.yml`:

```
  src_ip: id.orig_h
```

Which results in:
```
$ tools/sigmac -t splunk -c splunk-zeek rules/network/zeek/zeek_dce_rpc_domain_user_enumeration.yml
(sourcetype="bro:dce_rpc:json" (operation="LsarLookupNames3" OR operation="LsarLookupSids3" OR operation="SamrGetGroupsForUser" OR operation="SamrLookupIdsInDomain" OR operation="SamrLookupNamesInDomain" OR operation="SamrQuerySecurityObject" OR operation="SamrQueryInformationGroup")) | eventstats dc(operation) as val by id.orig_h | search val > 4
```

Anyways, I've extended the identifier token for aggregations [here](https://github.com/SigmaHQ/sigma/commit/068255fc82d45b019364018ae3475cfb19af467c).
